src/NServiceBus.Core/DelayedDelivery/TimeoutManager/TimeoutManager.cs

```diff
@@ -65,45 +65,50 @@ static void SetupTimeoutPoller(FeatureConfigurationContext context, string dispa
 
         static string SetupDispatcherSatellite(FeatureConfigurationContext context, string errorQueueAddress, TransportTransactionMode requiredTransactionSupport)
         {
-            string dispatcherAddress;
-            var dispatcherProcessorPipeline = context.AddSatellitePipeline("Timeout Dispatcher Processor", requiredTransactionSupport, PushRuntimeSettings.Default, "TimeoutsDispatcher", out dispatcherAddress);
-
-            dispatcherProcessorPipeline.Register("DispatchTimeoutRecoverability",
-                b => CreateTimeoutRecoverabilityBehavior(errorQueueAddress, dispatcherAddress, b),
-                "Handles failures when dispatching timeouts");
-
-            dispatcherProcessorPipeline.Register("TimeoutDispatcherProcessor", b => new DispatchTimeoutBehavior(
-                b.Build<IDispatchMessages>(),
-                b.Build<IPersistTimeouts>(),
-                requiredTransactionSupport),
-                "Terminates the satellite responsible for dispatching expired timeouts to their final destination");
-            return dispatcherAddress;
+            var instanceName = context.Settings.EndpointInstanceName();
+            var satelliteLogicalAddress = new LogicalAddress(instanceName, "TimeoutsDispatcher");
+            var satelliteAddress = context.Settings.GetTransportAddress(satelliteLogicalAddress);
+            var failureStorage = new TimeoutFailureInfoStorage();
+
+            context.AddSatelliteReceiver("Timeout Dispatcher Processor", satelliteAddress, requiredTransactionSupport, PushRuntimeSettings.Default, (builder, pushContext) =>
+            {
+                var recBehavior = CreateTimeoutRecoverabilityBehavior(errorQueueAddress, satelliteAddress, builder, failureStorage);
+                var dispatchBehavior = new DispatchTimeoutBehavior(
+                    builder.Build<IDispatchMessages>(),
+                    builder.Build<IPersistTimeouts>(),
+                    requiredTransactionSupport);
+
+                return recBehavior.Invoke(pushContext, () => dispatchBehavior.Invoke(pushContext));
```

Comment:
@timbussmann using the same reasoning (https://github.com/Particular/NServiceBus/pull/3797/files/b19da577bad60d80d9069beba417e8b38fef7cfd#r67942578) do you think we can call yagni on avoiding creating the behaviors every time since they will be gone very soon anyway? (I know we want to only merge high quality stuff so this is an honest question, on the fence my self)